compiler/rustc_expand/src/base.rs

```diff
@@ -1266,35 +1275,35 @@ pub fn expr_to_spanned_string<'a>(
                     "",
                     Applicability::MaybeIncorrect,
                 );
-                Some((err, true))
-            }
-            Ok(ast::LitKind::Err) => None,
-            Err(err) => {
-                report_lit_error(&cx.sess.parse_sess, err, token_lit, expr.span);
-                None
+                Ok((err, true))
             }
-            _ => Some((cx.dcx().struct_span_err(expr.span, err_msg), false)),
+            Ok(ast::LitKind::Err) => Err(cx
+                .dcx()
+                .span_delayed_bug(expr.span, "tried to get a string literal from `LitKind::Err`")),
```

Comment:
This delayed bug won't be necessary after https://github.com/rust-lang/rust/pull/121120.